sql/oracle_join.sql

```diff
@@ -35,12 +52,12 @@ EXPLAIN (COSTS off) SELECT t1.id, t2.id FROM typetest1 t1, typetest1 t2 WHERE t1
 EXPLAIN (COSTS off) SELECT t1.id, t2.id FROM typetest1 t1 JOIN typetest1 t2 ON t1.vc = t2.vc AND t1.lb = t2.lb;
 /* condition on one table needs to be evaluated locally */
 EXPLAIN (COSTS off) SELECT max(t1.id), min(t2.id) FROM typetest1 t1 JOIN typetest1 t2 ON t1.fl = t2.fl WHERE t1.vc || 'x' = 'shortx';
-/* outer joins */
-EXPLAIN (COSTS off) SELECT t1.id, t2.id FROM typetest1 t1 LEFT JOIN typetest1 t2 ON t1.d = t2.d;
-EXPLAIN (COSTS off) SELECT t1.id, t2.id FROM typetest1 t1 RIGHT JOIN typetest1 t2 ON t1.d = t2.d;
-EXPLAIN (COSTS off) SELECT t1.id, t2.id FROM typetest1 t1 FULL JOIN typetest1 t2 ON t1.d = t2.d;
 /* cross join */
 EXPLAIN (COSTS off) SELECT t1.id, t2.id FROM typetest1 t1 CROSS JOIN typetest1 t2;
+EXPLAIN (COSTS off) SELECT t1.id, t2.id FROM typetest1 t1 INNER JOIN typetest1 t2 ON true;
+EXPLAIN (COSTS off) SELECT t1.id, t2.id FROM typetest1 t1 LEFT  JOIN typetest1 t2 ON true;
+EXPLAIN (COSTS off) SELECT t1.id, t2.id FROM typetest1 t1 RIGHT JOIN typetest1 t2 ON true;
+EXPLAIN (COSTS off) SELECT t1.id, t2.id FROM typetest1 t1 FULL  JOIN typetest1 t2 ON true;
```

Comment:
I don't think we need four tests for that.

But we need more tests to test all branches. Some I can think of:

    SELECT a.c, b.nc FROM typetest1 a FULL JOIN (SELECT * FROM typetest1 WHERE id > 1) b USING (id);
    SELECT a.c, b.nc FROM typetest1 a LEFT JOIN (SELECT * FROM typetest1) b ON (a.id = b.id AND a.c >= b.c);
    SELECT a.c, b.nc FROM typetest1 a JOIN (SELECT * FROM typetest1) b ON (a.id = b.id AND a.c >= b.c);

Reply:
I'll rethink what test cases is needed to this patch.
Then, I'll add your test cases. :)